docs/csharp/language-reference/unsafe-code.md

```diff
@@ -13,90 +13,106 @@ helpviewer_keywords:
+- `int* p`: pointer to `int`
+- `int** p`: pointer to pointer to `int`
+- `int*[] p`: array of pointers to `int`
+- `char* p`: pointer to `char`
+- `void* p`: pointer to unknown type
 
-:::code language="csharp" source="snippets/unsafe-code/FixedKeywordExamples.cs" ID="5":::
+### Pointer operations
 
-You can't apply the indirection operator to a pointer of type `void*`. However, you can use a cast to convert a void pointer to any other pointer type, and vice versa.
+Pointers don't inherit from [`object`](builtin-types/reference-types.md). You can't box or unbox pointers, and there's no conversion between pointers and `object`. However, you can cast between pointer types and between pointers and integral types (with an explicit cast).
 
-A pointer can be `null`. Applying the indirection operator to a null pointer causes an implementation-defined behavior.
+The garbage collector doesn't track references from pointers. If you're pointing to a managed object, you must [pin](./statements/fixed.md) it for as long as the pointer is used.
 
-Passing pointers between methods can cause undefined behavior. Consider a method that returns a pointer to a local variable through an `in`, `out`, or `ref` parameter or as the function result. If the pointer was set in a fixed block, the variable to which it points might no longer be fixed.
-
-The following table lists the operators and statements that can operate on pointers in an unsafe context:
+The following operators and statements work with pointers in an unsafe context:
 
 | Operator/Statement                       | Use                                                            |
 |------------------------------------------|----------------------------------------------------------------|
-| `*`                                      | Performs pointer indirection.                                  |
-| `->`                                     | Accesses a member of a struct through a pointer.               |
-| `[]`                                     | Indexes a pointer.                                             |
-| `&`                                      | Obtains the address of a variable.                             |
-| `++` and `--`                            | Increments and decrements pointers.                            |
-| `+` and `-`                              | Performs pointer arithmetic.                                   |
-| `==`, `!=`, `<`, `>`, `<=`, and `>=`     | Compares pointers.                                             |
-| [`stackalloc`](operators/stackalloc.md)  | Allocates memory on the stack.                                 |
-| [`fixed` statement](statements/fixed.md) | Temporarily fixes a variable so that its address can be found. |
+| `*`                                      | Dereference (pointer indirection).                             |
+| `->`                                     | Access struct member through a pointer.                        |
```

Comment:
This isn't specific to structs. You can have say a `string* p` and do `p->Member()` as well.